### DISCUSSION TITLE
Не работает e621.net, хотя другие сайты работают.

### DISCUSSION THREAD

**Discussion by nkrapivin** [bol-van/zapret#991] [Q&A]
Конфиг: https://pastebin.com/Sk67iYGf
Результат blockcheck для e621.net: https://pastebin.com/ZfxFjMpc
IPv6 curl: https://pastebin.com/vg8819v6 (обрывается TLS хэндшейк)
IPv4 curl: https://pastebin.com/EfmMdVzx (подмена серта на `lawfilter.ertelecom.ru` и последующий облом)
Конфиг systemd-resolved: https://pastebin.com/gfmqbvCb
Ядро: `Linux Repka-Pi 5.19.16-sunxi64 #22.08.6 SMP Tue Oct 18 06:56:22 UTC 2022 aarch64 aarch64 aarch64 GNU/Linux`
ОС: Ubuntu 20.04 LTS ("Repka OS", от производителя одноплатника)

Почему-то открываются все заблокированные сайты (ну, что я знаю, рутрекер, дискорд и т.д.) но у e621 наблюдается странность.
Какое-то время подключения идут (у меня бот который стучит туда в апишку для получения постов), потом внезапно всё обрывается и ничего не помогает кроме перезагрузки всего одноплатника, что нежелательно, т.к. у меня там ещё несколько ботов и сайтик.
Причём т.к. этот сайт блокировали очень очень давно, в 2017 где-то, на DNS провайдера у этого сайта другие A и AAAA записи, которые редиректят тоже на этот lawfilter, но т.к. у меня DNS over TLS здесь подмена по DNS не срабатывает.
Может кто-нибудь подсказать в какую сторону глядеть? Я уже совсем не понимаю.

**Comment by numbereleven-a** [comment]
у него куча разных IP, может быть проблема в определённых? Можно попробовать дебаг глянуть в момент проблемы.
IP. не уверен что все верные
188.114.98.224
188.114.99.224
188.114.98.235
188.114.99.235
104.22.40.171
172.67.21.207
104.22.41.171
188.114.98.233
188.114.99.233

**Reply by nkrapivin** [reply]
Здравствуйте, спасибо за ваш ответ!
Там в логах curl видно, какой IP предпочитает на данный момент локальный резольвер.
Единственный вопрос, как мне заставить использование "другого" IP? /etc/hosts? Просто мне ещё важно чтобы из под httpx (httpx.AsyncTransport) это всё тоже работало, а у httpx не так много возможностей по кастомизации DNS резольвинга. Там по идее можно свой транспорт реализовать но я не особо углублялся, если честно.

**Reply by numbereleven-a** [reply]
hosts - да, про httpx не подскажу

**Reply by nkrapivin** [reply]
Попробовал подставить другой IP в хостс, но пока я точно не знаю решает ли это проблему, потому что я вот на 100% нашёл закономерность - сначала первые штук +-10 запросов идут нормально, но потом через какое-то время (+- полчаса) запросы вообще перестают идти, обрыв хэндшейка. И это точно не проблема самого сайта т.к. с компа и VPN всё открывается, ну и этот сайт как правило пишет авторам ботов на почту, если с их использованием API нашлись косяки.

**Comment by Fenrir-0xFF** [comment]
> Может кто-нибудь подсказать в какую сторону глядеть? Я уже совсем не понимаю.

Могу пока чисто по конфигу сказать, у тебя там 
TPWS_ENABLE=1
NFQWS_ENABLE=1
Запускаются одновременно tpws и nfqws, поставь TPWS_ENABLE=0 чтоб только nfqws запускался. 

Тут
#none,ipset,hostlist,autohostlist
MODE_FILTER=none
Если none то обход блокировок применяется вобще ко всем сайтам игноря <НOSTLIST> в стратегиях, желательно MODE_FILTER=autohostlist чтоб обход применялся только к сайтам которые есть у тебя в списке

Для HTTPS лучше используй стратегии которые с fake а не с syndata, у тебя syndata в основном на http в логе
